### PR TITLE
Minor Cleanup of CrystalFieldPeakBase

### LINKED ISSUE #15956: Minor Cleanup of Crystal Field Codes
In [CrystalFieldPeaks.cpp](Framework/CurveFitting/src/Functions/CrystalFieldPeaks.cpp):
* the objects/constants in the anonymous namespace should be treated as statics & their names in all CAPITALS to distinguish them from local variables
* `const bool real = true;`  & `const bool complx = true;` need could have better names to express their intent and again captalizing them
* it would be easier to read if all constants were defined together & then all anonymous functions and not a mixture.

Generally, `boost::lexical_cast` can now be replace with `std::to_string` to remove the boost dependency.

### PR DESCRIPTION
**Pull request by PranavBahuguna**
Refactor/cleanup of CrystalFieldPeakBase to improve clarity and readability.

**To test:**

Check all the code in [CrystalFieldPeaks.cpp](Framework/CurveFitting/src/Functions/CrystalFieldPeaks.cpp) to ensure that the issues of #15956 have been addressed.

Ensure any tests that utilise CrystalFieldPeakBase pass - these include:
- CrystalFieldMultiSpectrumTest
- CrystalFieldPeaksTest
- CrystalFieldSpectrumTest

Fixes #15956.

_Does not need to be in the release notes._
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
@PranavBahuguna Does this belong in `3.8` or is this a `3.9` maintenance issue?

**Comment by KarlPalmen** [comment]
I see it is now a Release 3.9 maintenance issue. :shipit:

**Comment by SimonHeybrock** [comment]
Not all of the items in the issue description seem to have been addressed? So the issue should not be closed (or the remaining items implemented)?

**Comment by KarlPalmen** [comment]
Can @SimonHeybrock be more specific, by stating which items have not been addressed?

**Comment by SimonHeybrock** [comment]
@KarlPalmen Sorry, I meant the last item (replacing `boost` with `std`).

**Comment by PranavBahuguna** [comment]
In CrystalFieldPeaksBase.cpp I did not see any instances of `boost` being used at all, so I didn't replace any.

File: Framework/CurveFitting/src/Functions/CrystalFieldPeaksBase.cpp [hunk 1]
```diff
@@ -65,17 +65,17 @@ void setSymmetryC2(API::IFunction &fun) {
   fun.unfixParameter("B40");
   fun.unfixParameter("B60");
   fixx(fun, "B21");
-  free(fun, "B22", real);
+  free(fun, "B22", REAL_PARAM_PART);
   fixx(fun, "B41");
-  free(fun, "B42", cmplx);
+  free(fun, "B42", CMPLX_PARAM_PART);
```
Comment (KarlPalmen) [inline Framework/CurveFitting/src/Functions/CrystalFieldPeaksBase.cpp]: Does this really mean imaginary part?

Comment (PranavBahuguna) [inline Framework/CurveFitting/src/Functions/CrystalFieldPeaksBase.cpp]: Terms seem nearly interchangeable, but I suppose imaginary makes more sense.

Comment (SimonHeybrock) [inline Framework/CurveFitting/src/Functions/CrystalFieldPeaksBase.cpp]: Complex = Real + Imaginary, so not at all interchangeable, i.e., :+1: for changing this.

**Comment by KarlPalmen** [comment]
Imaginary part is much clearer :+1:

**Comment by KarlPalmen** [comment]
Looking carefully at the code for the function `free()`, I see when it's 3rd argument is set to `REAL_PART`, then the real part is unfixed and the imaginary part is fixed at 0,0. If the 3rd argument is set to `IMAG_PART`, both parts are unfixed and the real part is not fixed to 0.0. This suggests to me that the original names were correct and just needed capitalising (without `PART`).

**Comment by SimonHeybrock** [comment]
@KarlPalmen Oops, I just merged this :(

**Pull request merged**